service/storage_service.cc

```diff
@@ -698,28 +699,123 @@ future<> storage_service::topology_transition() {
     _topology_state_machine.event.broadcast();
 }
 
+namespace {
+    future<> split_mutation(mutation source, std::vector<mutation>& target, size_t max_size) {
```

Comment:
> Why not return `future<std::vector<mutation>>` instead of output param?

It's more 'dependency inject-y'/'composable'  - e.g. we can send output mutations from several `split_mutation` calls into the same output vector, like we do here.

Reply:
> Also, maybe this is worth putting somewhere accessible, so other code can use it if needed.

I would be glad to, but I haven’t found a suitable place. Separate `.hh/.cc`?

Reply:
I'll put it into `mutation.hh/.cc`.

Reply:
I guess the best way would be to co_yield the mutations. This way it's clear it's an output parameter.

Reply:
e.g. seastar::experimental::generator

Reply:
seems too fancy